### PR TITLE
[Integration Tests] Cleanup usage of TestHelpers Part 3

### PR DESCRIPTION
**Pull request by fassadlr**

### CODE REVIEW

File: src/Stratis.Bitcoin.IntegrationTests/Mempool/MemoryPoolTests.cs [hunk 1]
```diff
@@ -21,21 +20,17 @@ public class MemoryPoolTests
...
         public void AddToMempool()
         {
             using (NodeBuilder builder = NodeBuilder.Create(this))
             {
-                CoreNode stratisNodeSync = builder.CreateStratisPowNode(this.network).NotInIBD().WithWallet();
-
-                builder.StartAll();
+                CoreNode stratisNodeSync = builder.CreateStratisPowNode(this.network).NotInIBD().WithWallet().Start();
 
                 TestHelper.MineBlocks(stratisNodeSync, 105); // coinbase maturity = 100
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests/Mempool/MemoryPoolTests.cs]: good call. Maybe we should call MineBlocks MineAndSync? Syncing is a bit of a hidden behavior.

File: src/Stratis.Bitcoin.IntegrationTests/Mempool/MemoryPoolTests.cs [hunk 2]
```diff
@@ -205,17 +189,15 @@ public void TxMempoolMapOrphans()
         {
             var rand = new Random();
             var randByte = new byte[32];
-            Func<uint256> randHash = () =>
+            uint256 randHash()
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests/Mempool/MemoryPoolTests.cs]: funky

File: src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletTests.cs [hunk 3]
```diff
@@ -67,7 +62,7 @@ public void WalletCanReceiveAndSendCorrectly()
                 Assert.Null(stratisReceiver.FullNode.WalletManager().GetSpendableTransactionsInWallet(WalletName).First().Transaction.BlockHeight);
 
                 // Generate two new blocks so the transaction is confirmed
-                TestHelper.GenerateBlockManually(stratisSender, new List<Transaction>(new[] { stratisSender.FullNode.Network.CreateTransaction(trx.ToBytes()) }));
+                stratisSender.AddToStratisMempool(trx);
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletTests.cs]: Is that the same behavior?

File: src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletTests.cs [hunk 4]
```diff
@@ -88,41 +83,33 @@ public void WalletCanReorg()
...
+                CoreNode stratisReceiver = builder.CreateStratisPowNode(this.network).NotInIBD().WithWallet().Start();
+                CoreNode stratisReorg = builder.CreateStratisPowNode(this.network).NotInIBD().WithWallet().Start();
 
                 int maturity = (int)stratisSender.FullNode.Network.Consensus.CoinbaseMaturity;
                 TestHelper.MineBlocks(stratisSender, maturity + 15);
 
                 int currentBestHeight = maturity + 15;
 
-                // Wait for block repo for block sync to work
-                TestHelper.WaitLoop(() => TestHelper.IsNodeSynced(stratisSender));
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletTests.cs]: Not needed?

File: src/Stratis.Bitcoin.IntegrationTests/Mempool/MemoryPoolTests.cs (hunk 1 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests/Mempool/MemoryPoolTests.cs]: OK yeah we can rename in the next PR?

File: src/Stratis.Bitcoin.IntegrationTests/Mempool/MemoryPoolTests.cs (hunk 2 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests/Mempool/MemoryPoolTests.cs]: Yeah this was suggested by visual studio ;)  to use a local function

File: src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletTests.cs (hunk 3 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletTests.cs]: Kind of yes, there is no difference between generating a block manually and including a transaction as opposed to just adding the tx to the mempool and mine it normally.

File: src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletTests.cs (hunk 4 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletTests.cs]: Nope this is done in `TestHelper.MineBlocks`

File: src/Stratis.Bitcoin.IntegrationTests/Mempool/MemoryPoolTests.cs (hunk 1 above)
Comment (bokobza) [reply src/Stratis.Bitcoin.IntegrationTests/Mempool/MemoryPoolTests.cs]: sure

File: src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletTests.cs (hunk 3 above)
Comment (bokobza) [reply src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletTests.cs]: ok sounds good

File: src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletTests.cs (hunk 4 above)
Comment (bokobza) [reply src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletTests.cs]: ah yes

**Comment by bokobza** [comment]
Good work mate.

**Pull request merged**